mbed_app.json

```
@@ -1,15 +1,60 @@
 {
+    "config": {
+        "demo-button": {
+            "macro_name": "DEMO_BUTTON",
+            "required": true
+        },
+        "demo-button-active-low": {
+            "help": "true if the button state is low when pressed, high when released",
+            "macro_name": "DEMO_BUTTON_ACTIVE_LOW",
+            "required": false
+        }
+    },
     "target_overrides": {
         "*": {
             "mcuboot.bootloader-build": 0,
-            "target.mbed_app_start": "0x20000",
-            "target.mbed_app_size": "0xDC000"
+            "target.c_lib": "small",
+            "mcuboot.log-level": "MCUBOOT_LOG_LEVEL_DEBUG",
+            "mbed-trace.enable": true
         },
         "NRF52840_DK": {
-            "mcuboot.max-img-sectors": 256
+            "demo-button": "BUTTON1",
+            "demo-button-active-low": true,
+            "target.mbed_app_start": "0x21000",
+            "target.mbed_app_size": "0xBE000",
+            "mcuboot.primary-slot-address": "0x20000",
+            "mcuboot.slot-size": "0xC0000",
+            "mcuboot.scratch-address": "0xE0000",
+            "mcuboot.scratch-size": "0x20000",
+            "mcuboot.max-img-sectors": "0x180",
+            "mcuboot.read-granularity": 4,
+            "qspif.QSPI_MIN_PROG_SIZE": 4
         },
         "EP_AGORA": {
-            "mcuboot.max-img-sectors": 256
+            "demo-button": "BUTTON1",
+            "demo-button-active-low": true,
+            "target.mbed_app_start": "0x21000",
+            "target.mbed_app_size": "0xBE000",
+            "mcuboot.primary-slot-address": "0x20000",
+            "mcuboot.slot-size": "0xC0000",
+            "mcuboot.scratch-address": "0xE0000",
+            "mcuboot.scratch-size": "0x20000",
+            "mcuboot.max-img-sectors": "0x180",
+            "mcuboot.read-granularity": 4,
+            "qspif.QSPI_MIN_PROG_SIZE": 4
+        },
+        "DISCO_L475VG_IOT01A": {
+            "demo-button": "USER_BUTTON",
+            "demo-button-active-low": true,
+            "target.mbed_app_start": "0x8021000",
+            "target.mbed_app_size": "0xBE000",
+            "mcuboot.primary-slot-address": "0x8020000",
+            "mcuboot.slot-size": "0xC0000",
+            "mcuboot.scratch-address": "0x80E0000",
+            "mcuboot.scratch-size": "0x20000",
+            "mcuboot.max-img-sectors": "0x180",
+            "mcuboot.read-granularity": 4,
+            "qspif.QSPI_MIN_PROG_SIZE": 4
```

```suggestion
            "mcuboot.read-granularity": 1,
            "qspif.QSPI_MIN_PROG_SIZE": 1
```
The Disco board allows one-byte access. See https://github.com/ARMmbed/mbed-os/pull/13848/files